### PR TITLE
upgrade golang to 1.9.3 to solve compile issue

### PR DESCRIPTION
**Pull request by figo**
otherwise, an error pop up:
vendor/github.com/json-iterator/go/config.go:107: undefined: sync.Map

Signed-off-by: Hui Luo <luoh@vmware.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@ endif
 PKGS=$(shell go list ./... | grep -v /vendor)
 CI_PKGS=$(shell go list ./... | grep -v /vendor | grep -v test)
 FMT_PKGS=$(shell go list -f {{.Dir}} ./... | grep -v vendor | grep -v test | tail -n +2)
-SHELL_IMAGE=golang:1.8.3
+SHELL_IMAGE=golang:1.9.3
```
Comment (detiber) [inline Makefile]: Would this build with a newer version of golang or is`1.9.3` the latest that will work without changes?

File: Makefile [hunk 2]
```diff
@@ -52,7 +52,7 @@ build-linux-amd64: ## Create the kubicorn executable for Linux 64-bit OS in the
 	-w /go/src/github.com/kubicorn/kubicorn \
 	-v ${PWD}:/go/src/github.com/kubicorn/kubicorn \
 	-e GOPATH=/go \
-	--rm golang:1.8.1 make docker-build-linux-amd64
+	--rm golang:1.9.3 make docker-build-linux-amd64
```
Comment (detiber) [inline Makefile]: Should this possible reference the SHELL_IMAGE variable instead of hardcoding it here?

File: Makefile (hunk 1 above)
Comment (figo) [reply Makefile]: i did not test further, 1.9.* is probably the minimum upgrade to pass compiling, the e2e test based on kubicorn binary (not in this repo) passed after this change.

also, there is a discussion around this seems to confirm the theory:
https://github.com/kubernetes/test-infra/issues/7183

File: Makefile (hunk 2 above)
Comment (figo) [reply Makefile]: sounds good, i'll change it

**Pull request merged**